TCAT/Views/WhatsNewHeaderView.swift

```diff
@@ -78,65 +84,84 @@ class WhatsNewHeaderView: UIView {
             make.centerX.equalToSuperview()
             make.height.equalTo(whatsNewHeader.intrinsicContentSize.height)
             make.width.equalTo(whatsNewHeader.intrinsicContentSize.width)
         }
 
         updateTitle.snp.makeConstraints { (make) in
-            make.top.equalTo(whatsNewHeader.snp.bottom).offset(8)
+            updateNameToTitle = make.top.equalTo(whatsNewHeader.snp.bottom).offset(8).constraint
             make.centerX.equalToSuperview()
             make.height.equalTo(updateTitle.intrinsicContentSize.height)
             make.width.equalTo(updateTitle.intrinsicContentSize.width)
         }
 
         updateDescription.snp.makeConstraints { (make) in
             let value = CGFloat(32)
-            make.top.equalTo(updateTitle.snp.bottom).offset(6)
-            make.centerX.equalToSuperview()
+            updateDescToUpdateName = make.top.equalTo(updateTitle.snp.bottom).offset(6).constraint
             make.leading.trailing.equalToSuperview().inset(value)
-            if
-                let containerView = superview,
-                let tableView = containerView.superview,
-                let wholeView = tableView.superview,
-                let description = updateDescription.text
+            if let description = updateDescription.text
             {
                 // Take total width and subtract various insets used in layout
                 let headerViewCardPadding = containerPadding.left + containerPadding.right
-                let widthValue = wholeView.frame.width - headerViewCardPadding - (value * 2)
+                let widthValue = UIScreen.main.bounds.width - headerViewCardPadding - (value * 2)
                 
                 let heightValue = ceil(description.heightWithConstrainedWidth(width: widthValue, font: updateDescription.font))
+                updateDescriptionHeight = ceil(heightValue)
                 make.height.equalTo(ceil(heightValue))
             }
         }
 
         dismissButton.snp.makeConstraints { (make) in
-            make.top.equalTo(updateDescription.snp.bottom).offset(12)
+            dismissButtonToUpdateDesc = make.top.equalTo(updateDescription.snp.bottom).offset(12).constraint
             make.centerX.equalToSuperview()
             make.width.equalTo(90)
-            make.height.equalTo(dismissButton.intrinsicContentSize.height)
+            dismissButtonToBottom = make.bottom.equalToSuperview().inset(16).constraint
         }
-
-        snp.makeConstraints { (make) in
-            make.bottom.equalTo(dismissButton.snp.bottom).offset(16)
+    }
+    
+    func calculateCardHeight() -> CGFloat {
+        if let titleToTop = titleToTop, let updateNameToTitle = updateNameToTitle, let updateDescToUpdateName = updateDescToUpdateName, let dismissButtonToUpdateDesc = dismissButtonToUpdateDesc, let dismissButtonToBottom = dismissButtonToBottom {
+            
+            let titleToTopVal = titleToTop.layoutConstraints[0].constant
```

Comment:
is this a safe value to access? Can this ever fail?

Reply:
With how I've structured it, It shouldn't because if its not nil, then there exists a constraint and layoutConstraints[0] will always have something to access